crates/matrix-sdk-crypto/src/types/events/olm_v1.rs

```diff
@@ -169,6 +176,22 @@ impl<C: EventType + Debug + Sized + Serialize> DecryptedOlmV1Event<C> {
     }
 }
 
+/// A decrypted to-device "m.dummy" event.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ToDeviceDummyEvent {
```

Comment:
This defines a whole struct for the `m.dummy` event. What we instead want is to reuse the generic struct for decrypted events we have:

https://github.com/matrix-org/matrix-rust-sdk/blob/63c8696cac91a2dd1e941be5a66fceb59f3b15bb/crates/matrix-sdk-crypto/src/types/events/olm_v1.rs#L143-L157

We'll have to define the `m.dummy` content so it can fit into the generic `C` variable for the struct from above.

Reply:
Hey, I just had an idea: what if instead of everything we just split the final match block in `machine.rs` and use something like this instead:

```rust
            AnyDecryptedOlmEvent::SecretSend(e) => {
                // ...
            }
            AnyDecryptedOlmEvent::Custom(e) => match e.event_type {
                "m.dummy" => {
                    warn!("Received an unexpected encrypted to-device event");
                }
                _ => {
                    warn!("Received an unexpected encrypted to-device event");
                }
            },
```

Will that work?

Reply:
I would also work, but deserializing into a known type is likely more performant then going into the `Custom` branch. 

And having strong types for all the known and used things is much more preferable then matching things by string.